### PR TITLE
fulfil: instructors section completed and created docs for students-p…

### PR DESCRIPTION
**Pull request by MubiynDev**
…rojects domain

### CODE REVIEW

**Comment by MubiynDev** [comment]
fulfil: instructors section completed and created docs for students-projects domain

File: src/domain/instructors/instructor.repo.ts [hunk 1]
```diff
@@ -2,23 +2,35 @@ import { instructorModel } from "@/domain/instructors/instructor.model"
...
 
-    public getById = async (id: string): Promise<InstructorDto> => {
+    getById = async (id: string): Promise<InstructorDto> => {
         return this.instructorModel.findById(id);
       };
 
-    public getByfirstName = async (firstName: string): Promise<InstructorDto> => {
-        return this.instructorModel.findById(firstName);
-      };
+    getByfirstName = async (firstName: string): Promise<InstructorDto> => {
```
Comment (azoirov) [inline src/domain/instructors/instructor.repo.ts]: remove if not used

File: src/domain/instructors/instructor.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import { CreateInstructorDto, InstructorDto } from "./dto/instructor.dto";
+import InstructorRepo from "./instructor.repo";
+
+class InstructorService {
+    public InstructorRepo = new InstructorRepo()
+
+    public create = async (data: CreateInstructorDto): Promise<InstructorDto> => {
+        const instructor = await this.InstructorRepo.getByPhone(data.phone)
+
+        if(instructor) throw new BadRequestError(ErrorCode.InstructorAlreadyExists)
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts]: user ConflictError

File: src/domain/instructors/instructor.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import ErrorCode from "@/enums/error-code.enum";
+import BadRequestError from "@/errors/bad-request.error";
+import NotFoundError from "@/errors/not-found.error";
+import { CreateInstructorDto, InstructorDto } from "./dto/instructor.dto";
+import InstructorRepo from "./instructor.repo";
+
+class InstructorService {
+    public InstructorRepo = new InstructorRepo()
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts]: public instructorRepo = new InstructorRepo()

File: src/domain/instructors/instructor.service.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    public create = async (data: CreateInstructorDto): Promise<InstructorDto> => {
+        const instructor = await this.InstructorRepo.getByPhone(data.phone)
+
+        if(instructor) throw new BadRequestError(ErrorCode.InstructorAlreadyExists)
+
+        return this.InstructorRepo.create(data)
+    }
+
+    public getAll = async ():Promise<InstructorDto[]>  => {
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts:19]: return this.instructorRepo.getAll()

File: src/domain/instructors/instructor.service.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        const result = await this.InstructorRepo.getAll()
+        return result;
+    }
+
+    public getById = async (id: string):Promise<InstructorDto>  => {
+        const result = await this.InstructorRepo.getById(id)
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts]: const instructor = await this.instructorRepo.getById(id);

if(!instructor) throw new NotFoundError(ErrorCode.InsturctorNotFound)

return instructror

File: src/domain/instructors/instructor.service.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    public update = async (
+        id: string,
+        updateData: Partial<InstructorDto>
+      ): Promise<InstructorDto> => {
+        const instructor = await this.InstructorRepo.getById(id);
+        if (!instructor) throw new NotFoundError(ErrorCode.SpNotFound);
+    
+        const result = await this.InstructorRepo.update(id, updateData);
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts]: return this.intructorRepo.update(....)

File: src/domain/instructors/instructor.service.ts [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        return result;
+      };
+
+    public deleteById = async (id: string): Promise<InstructorDto> => {
+        const instructor = await this.InstructorRepo.getById(id);
+        if (!instructor) throw new NotFoundError(ErrorCode.InstructorNotFound);
+    
+        const result = await this.InstructorRepo.deleteById(id);
```
Comment (azoirov) [inline src/domain/instructors/instructor.service.ts]: return this.instructorRepo.....

**Review by azoirov — CHANGES_REQUESTED** [review]

File: src/domain/instructors/instructor.service.ts (hunk 2 above)
Comment (azoirov) [reply src/domain/instructors/instructor.service.ts]: bu ozgarmabti

Comment (MubiynDev) [reply src/domain/instructors/instructor.service.ts]: togrladim

**Review by azoirov — APPROVED** [review]

**Pull request merged**